### PR TITLE
Reenable mock server

### PR DESCRIPTION
**Pull request by Osmose**
This is #434, with the commit from #440 that added a wait script for docker containers to fix mock-recipe-server builds, along with another commit that fixes a bug in container cleanup (the commit message explains it well) and re-enables mock-recipe-server in CI.

This is blocked by #434, and I'll rebase it after that lands. But I want to put it up now to see what automation thinks of it. If automation is happy, this could probably be reviewed as well, if anyone wants to take a glance at the two commits directly. Otherwise, once I rebase I'll request reviews myself.

### CODE REVIEW

**Comment by rehandalal** [comment]
you're missing https://github.com/mozilla/normandy/pull/434/commits/45ddb78b6eaafbf5a48a48aff4f4e672b5a3450b from #434, so tests won't pass.

**Comment by Osmose** [comment]
> you're missing 45ddb78 from #434, so tests won't pass.

Oh, good point, thanks! I was using the branch from your fork rather than the one on the main repo.

File: mock-recipe-server/bin/generate.sh [hunk 1]
```diff
@@ -3,16 +3,26 @@ set -eu
...
+    docker-compose \
+        -p mockrecipeserver \
+        -f "$REPO_DIR/compose/docker-compose.yml" \
+        -f "$REPO_DIR/mock-recipe-server/docker-compose.yml" \
+        $@
+}
+
+# Shut down docker even if we error out.
+function finish {
+    compose down
```
Comment (mythmon) [inline mock-recipe-server/bin/generate.sh]: I think we only need to run `compose stop`. We can't actually remove images in CircleCI, and I think it is probably fine to leave things there but not running.

**Comment by Osmose** [comment]
Yesssssssssss

Turns out I was missing @mythmon's fix for building using the `mozilla/normandy:latest` tag on this particular branch. The build passes!

**Comment by rehandalal** [comment]
🔥🔥

**Pull request merged**